examples/moving_least_squares/moving_least_squares.cpp

```diff
@@ -0,0 +1,335 @@
+/****************************************************************************
+ * Copyright (c) 2017-2023 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+// Example taken from DataTransferKit
+// (https://github.com/ORNL-CEES/DataTransferKit)
+// with MLS resolution from
+// (http://dx.doi.org/10.1016/j.jcp.2015.11.055)
+
+#include <ArborX.hpp>
+#include <Kokkos_Core.hpp>
+
+#include <limits>
+
+using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+using MemorySpace = ExecutionSpace::memory_space;
+
+struct RBFWendland_0
+{
+  KOKKOS_INLINE_FUNCTION double operator()(double x)
+  {
+    x /= _radius;
+    return (1. - x) * (1. - x);
+  }
+
+  double _radius;
+};
+
+struct MVPolynomialBasis_3D
+{
+  static constexpr std::size_t size = 4;
+
+  template <typename Double3D>
+  KOKKOS_INLINE_FUNCTION Kokkos::Array<double, size>
+  operator()(Double3D const &p) const
+  {
+    return {{1., p[0], p[1], p[2]}};
+  }
+};
+
+// Func to evaluate
+template <typename Double3D>
+KOKKOS_INLINE_FUNCTION double func(Double3D const &p) {
+  return Kokkos::sin(p[0]) + Kokkos::cos(p[1]) + p[2];
+} 
+
+int main(int argc, char *argv[])
+{
+  Kokkos::ScopeGuard guard(argc, argv);
+  constexpr std::size_t num_neighbors = 10;
+  constexpr std::size_t source_points_num = 10;
+  constexpr std::size_t target_points_num = 4;
+
+  auto source_points = Kokkos::View<ArborX::Point *, MemorySpace>(
+    "source_points", source_points_num);
+  auto target_points = Kokkos::View<ArborX::Point *, MemorySpace>(
+    "target_points", target_points_num);
+  auto source_points_host = Kokkos::create_mirror_view(source_points);
+  auto target_points_host = Kokkos::create_mirror_view(target_points);
+
+  // Generate source points
```

Comment:
It would help to provide a description or an ascii image of how the points are distributed.

Reply:
I will change how points are generated as this was only the basis to have the minimum code required to be able to test the example. I'll probably generate them to be as a mesh inside a cube.